components/base/kinematicbase/fake_kinematics.go

```diff
@@ -127,6 +137,46 @@ type fakePTGKinematics struct {
 	sleepTime   int
 }
 
+// NewPTGFrameFromKinematicOptions returns a new PTGFrame based on the properties & geometries of the base.
+func NewPTGFrameFromKinematicOptions(
```

Comment:
This is duplicating code from the other constructor and basically does nothing on top of the existing function in tpspace package except allowing the use of the Options parameter.  I tried doing some reorganizing and landed on something I think would be a good improvement.  Make `PTGOptions` public in the tpspace package and have the kinematic base options take those as an argument in the constructor.  Then the test code can instantiate the frame from these options, and pass the complete frame into the constructor.